src/Vcs/Git.php

```diff
@@ -130,6 +129,21 @@ public function getPreReleasesForVersion(string $version): array
         return array_map([$this, 'getVersionFromTag'], $tags);
     }
 
+    /**
+     * @param mixed[] $output
+     * @SuppressWarnings(PHPMD.ExitExpression)
+     */
+    protected static function checkExitCode(int $exitCode, array $output): void
+    {
+        switch ($exitCode) {
+            case 128:
+                echo('Error: Repository not found in current path.' . PHP_EOL);
+                exit(1);
+            default:
+                throw new GitException(implode(PHP_EOL, $output));
+        }
+    }
+
```

Comment:
Exit should almost never be used except at the highest level of a script when it is specifically a CLI script.
Since this project uses Symfony Console that part is abstracted away hence exit should NEVER be used.


Failing with a stack trace is not really a problem, it just shouldn't ever be reached in a normal path of operation.

Better would be to validate the user input (the current working directory is just some user input) and give a proper
error at the command level if the user input is invalid.

Reply:
But the error, or exception, doesn't occur during a 'normal' command execution. It occurs during application startup so to speak, so there isn't even a `$output OutputInterface` at hand to use.